--- 1330_slow.hs	2013-01-27 07:01:28.000000000 +0600
+++ 1330.hs	2013-01-27 07:00:01.000000000 +0600
@@ -1,17 +1,22 @@
 import Control.Applicative
 import Control.Monad
 import Data.Array
+import Data.Maybe
 import Data.Time.Clock
 import System.IO
 import Text.Printf
 
+import qualified Data.ByteString.Char8 as B
+
+int = fst . fromJust . B.readInt
+
 readInt :: IO Int
-readInt = readLn
+readInt = int <$> B.getLine
 
 readIntPair :: IO [Int]
-readIntPair = parseLine <$> getLine
+readIntPair = parseLine <$> B.getLine
     where
-        parseLine = (map read) . words
+        parseLine = (map int) . B.words
 
 getAnswer prefixSums (from:to:_) = (prefixSums ! to) - (prefixSums ! (from - 1))
 
